docker/install.sh

```diff
@@ -14,22 +14,31 @@
 #  License for the specific language governing permissions and limitations under
 #  the License.
 
-# Source ros2
-if [[ ! -z "$ROS2_PACKAGES" ]]; then
+if [[ ! -z "$PACKAGES" ]]; then
+    # No need to install and build dependencies as they are already present in /opt/carma/install/setup.bash
     echo "Sourcing previous build for incremental build start point..."
     source /opt/carma/install/setup.bash
 else
     echo "Sourcing base image for full build..."
-    source /opt/ros/foxy/setup.bash
-fi
+    source /opt/ros/humble/setup.bash
+    source /opt/autoware.ai/ros/install/setup.bash
+
+    # Get driver
+    sudo apt-get update && sudo apt install curl gnupg2 lsb-release
+    sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key  -o /usr/share/keyrings/ros-archive-keyring.gpg
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
 
-# Get driver
-sudo apt-get update && sudo apt install curl gnupg2 lsb-release
-sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key  -o /usr/share/keyrings/ros-archive-keyring.gpg
-echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+    # Before installing driver and the wrapper, temporary workaround until Eigen3 and PCL are properly exported by velodyne_pointcloud
+    # open issue: https://github.com/ros-drivers/velodyne/issues/550, workaround by: @hect95
+    sudo sed -i 's/\beigen\b/Eigen3/g' /opt/ros/humble/share/velodyne_pointcloud/cmake/ament_cmake_export_dependencies-extras.cmake
+    sudo sed -i 's/\bpcl\b/PCL/g' /opt/ros/humble/share/velodyne_pointcloud/cmake/ament_cmake_export_dependencies-extras.cmake
+    
+    # Install the driver and pointcloud conversion package
+    sudo apt install ros-humble-velodyne-driver -y
 
-# Install the driver and pointcloud conversion package
-sudo apt install ros-foxy-velodyne-driver -y
+    source /opt/ros/humble/setup.bash
+    source /opt/autoware.ai/ros/install/setup.bash
```

Comment:
We shouldn't need to source here since we already do one lines 23-24

Reply:
Don't we need to source after installing the package? Or does it source automatically

Reply:
Or I guess we should only source here and not above

Reply:
Yeah good point, should just need to remove the above source then